packages/react-charting/src/components/SankeyChart/SankeyChart.test.tsx

```diff
@@ -0,0 +1,464 @@
+    ],
+  },
+};
+
+describe('Sankey Chart snapShot testing', () => {
+  it('renders Sankey correctly', () => {
+    const component = renderer.create(<SankeyChart data={data} height={500} width={800} />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders Sankey correctly on providing nodecolors and border colors ', () => {
+    const nodeColors = ['#E3008C', '#00A2AD', '#022F22', '#00188F'];
+    const borderColors = ['#002E39', '#43002C', '#3B52B4'];
+
+    const component = renderer.create(
+      <SankeyChart
+        data={dataWithoutColors}
+        height={500}
+        width={800}
+        colorsForNodes={nodeColors}
+        borderColorsForNodes={borderColors}
+      />,
+    );
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe('Render calling with respective to props', () => {
+  it('No prop changes', () => {
+    const renderMock = jest.spyOn(SankeyChartBase.prototype, 'render');
+    const props = {
+      data,
+      height: 500,
+      width: 800,
+    };
+    const component = mount(<SankeyChart {...props} />);
+    component.setProps({ ...props });
+    expect(renderMock).toHaveBeenCalledTimes(2);
+    renderMock.mockRestore();
+  });
+
+  it('prop changes', () => {
+    const renderMock = jest.spyOn(SankeyChartBase.prototype, 'render');
+    const props = {
+      data,
+      height: 700,
+      width: 1100,
+    };
+    const component = mount(<SankeyChart {...props} />);
+    component.setProps({ ...props, height: 1000 });
+    expect(renderMock).toHaveBeenCalledTimes(2);
+    renderMock.mockRestore();
+  });
+});
+
+describe('SankeyChart - mouse events', () => {
+  beforeEach(sharedBeforeEach);
+  afterEach(sharedAfterEach);
+
+  it('Should render correctly on node mouseover', () => {
+    wrapper = mount(<SankeyChart data={data} height={500} width={800} />);
+    wrapper.find('rect').at(1).simulate('mouseover');
+    const tree = toJson(wrapper, { mode: 'deep' });
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('Should render correctly on link mouseover', () => {
+    wrapper = mount(<SankeyChart data={data} height={500} width={800} />);
+    wrapper.find('path').at(1).simulate('mouseover');
+    const tree = toJson(wrapper, { mode: 'deep' });
+    expect(tree).toMatchSnapshot();
+  });
+  it('Should render callout correctly on mouseover when height of node is less than 24px', () => {
+    wrapper = mount(<SankeyChart data={data} height={500} width={800} />);
+    wrapper.find('rect[id="nodeBar12"]').at(0).simulate('mouseover');
+    const tree = toJson(wrapper, { mode: 'deep' });
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('Should render tooltip correctly on mouseover when node description is large', () => {
+    wrapper = mount(<SankeyChart data={data} height={500} width={800} />);
+    wrapper.find('text[id="nodeBar48-name"]').at(0).simulate('mouseover');
+    const tree = toJson(wrapper, { mode: 'deep' });
```

Comment:
Is tooltip rendered here. Can you add an explicit check for the tooltip.

Reply:
This is done for the data in which a node has a truncated text, so tooltip check is there in this.